src/makielayout/blocks/menu.jl

```diff
@@ -202,41 +179,58 @@ function initialize_block!(m::Menu; default = nothing)
         )
     end
 
-    onmouseover(mouseevents) do me
-        i = pick_entry(me)
-        optionpolycolors[] = map(eachindex(optionstrings[])) do j
-            j == m.i_selected[] ? m.cell_color_active[] :
-            i == j ? m.cell_color_hover[] :
-                iseven(i) ? to_color(m.cell_color_inactive_even[]) :
-                to_color(m.cell_color_inactive_odd[]) 
-        end
-        return Consume(false)
-    end
-
-    onmouseout(mouseevents) do me
-        optionpolycolors[] = map(eachindex(optionstrings[])) do i
-            i == m.i_selected[] ? m.cell_color_active[] :
-            iseven(i) ? to_color(m.cell_color_inactive_even[]) :
-                to_color(m.cell_color_inactive_odd[]) 
-        end
-        return Consume(false)
-    end
-
-    onmouseleftclick(mouseevents) do me
-        i = pick_entry(me)
-        m.i_selected[] = i
-        m.is_open[] = false
-        return Consume(true)
-    end
-
-    # To stop other things from triggering on left down inside the dropdown menu
-    onmouseleftdown(_ -> Consume(true), mouseevents)
-
-    # close the menu if the user clicks somewhere else
-    # the logic is a bit convoluted because menuscene and selection poly have to be checked at the same time
-    on(blockscene.events.mousebutton, priority=61) do butt
-        if butt.action === Mouse.release && !is_mouseinside(menuscene) && !(mouseposition_px(blockscene) in selectionarea[])
-            m.is_open[] = false
+    was_inside_options = false
+    was_inside_button = false
+    e = menuscene.events
+    onany(e.mouseposition, e.mousebutton, priority=typemax(Int)) do position, butt
+        mp = screen_relative(menuscene, position)
+        if Makie.is_mouseinside(menuscene)
+            is_over = mouseover(menuscene, optionpolys, optiontexts)
+            pressed = ispressed(menuscene, Mouse.left)
+            if is_over
+                was_inside_options = true
+                if pressed
```

Comment:
I don't think this matters/happens in practice but `ispressed` could technically trigger multiple times because it's true as long as the left mouse button is held down. It's also a bit different from previous behavior which reacted to left-mouse-up on an option instead of mouse down (or technically any action while left-mouse-down) if I'm not mistaken.

`if butt.button == Mouse.left && butt.action = Mouse.up` should match master behavior a bit better. Might also be slightly faster, though irrelevant compared to picking.

Reply:
Ah good point, I think it does actually matter, since I noticed that sometimes it seemed to press 2 times, so it closed the menu again!

Reply:
Yea I think that's happening in the one below